fly/src/semantics.rs

```diff
@@ -407,6 +354,68 @@ impl Model {
     }
 }
 
+impl std::fmt::Display for Model {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        fn fmt_rel(decl: &RelationDecl, interp: &Interpretation) -> String {
+            fn fmt_element(sort: &Sort, element: Element) -> String {
+                match sort {
+                    Sort::Bool => match element {
+                        0 => "false".to_string(),
+                        1 => "true".to_string(),
+                        _ => unreachable!(),
+                    },
+                    Sort::Uninterpreted(s) => format!("@{s}_{element}"),
+                }
+            }
+
+            let mut lines = vec![];
+            let args_list = interp.shape[..interp.shape.len() - 1]
+                .iter()
+                .map(|&card| (0..card).collect::<Vec<Element>>())
+                .multi_cartesian_product_fixed()
+                .collect::<Vec<_>>();
+            for args in args_list {
+                let name = &decl.name;
+                let args_s = decl
+                    .args
+                    .iter()
+                    .zip(&args)
+                    .map(|(typ, &idx)| fmt_element(typ, idx))
+                    .collect::<Vec<_>>();
+                let args_s = if args_s.is_empty() {
+                    "".to_string()
```

Comment:
I prefer to write this with `args_s == ""` and `format!("")` (overriding clippy warnings for both).

Reply:
`args_s` is a vector, not a string.

Reply:
ah I see, I was confused because it's shadowed with a different type